src/BinaryParsers/ElfBinary/Dwarf/DwarfMemoryReader.cs

```diff
@@ -249,6 +249,11 @@ public byte[] ReadBlock(uint size)
         /// <param name="position">The position.</param>
         public byte[] ReadBlock(uint size, int position)
         {
+            if (position >= Data.Length)
```

Comment:
>position >= Data.Length [](http://example.com/codeflow?start=16&length=23)

is it possible to create a binary to test this cases?

Reply:
yes! have no idea currently but let me check

Reply:
not able to get the exact same issue but I found another issue which is similar out of boundary. it is when compiled with special machine type.
fixed the error and added 2 test files.